WordPress/performance
# Implement logic to measure specific hook execution time with Server-Timing controlled by a WP Admin screen

## Summary

This PR implements a WP Admin screen _Tools > Server-Timing_ which allows providing action and filter names. Any hook names provided will be measured and exposed in the `Server-Timing` header in the frontend.

While the Server-Timing API from #553 has been a powerful foundation, it can sometimes be cumbersome to measure specific hooks performance, since it always requires writing custom code. [Gists like this one](https://gist.github.com/felixarntz/63c05392dbf7d51cc7f8f4a424b1ff39) can be used for that, but even with such thing it may be easier to just add the action or filter to a UI instead of having to modify a plugin's code. This PR makes it really easy and gives site owners control about specific hooks to measure.

<img width="1186" alt="Admin screen to manage hooks to measure with Server-Timing" src="https://github.com/WordPress/performance/assets/3531426/2d897138-894b-4a35-b02e-e2a6bb3bec54">

_The new admin screen with example input_

## Relevant technical choices

* Adds a new WordPress option/setting to store Server-Timing configuration.
* Adds a new WP Admin screen with textareas to provide action and filter names.
* Uses the data from the new option/setting to measure the duration of those hooks, using similar logic as previously implemented in [the aforementioned Gist](https://gist.github.com/felixarntz/63c05392dbf7d51cc7f8f4a424b1ff39).

## Testing
1. Go to _Tools > Server-Timing_.
2. Add some hook names (actions and/or filters), one per line. For example those from the screenshot above.
3. Save and verify the hook names are still shown in the textareas, in a sanitized way.
    * For example, try to add spaces or (other) invalid characters and check that they're stripped after submitting.
4. Load the frontend.
5. Type `console.table( performance.getEntries()[0].serverTiming )` in the browser's dev tools console.
6. Verify that you see measurements for the actions/filters you provided, via `wp-action-{$action}` / `wp-filter-{$filter}` metrics.
    * Note that metrics will only show up if the respective action/filter was actually run during the request.
    * If you provide hook names that are only run after the template is being parsed (e.g. `wp_enqueue_scripts` action), they won't show up by default. You'll need to activate output buffering (e.g. via [this Gist plugin](https://gist.github.com/felixarntz/9c3d7150c74082e69bb426393b68b12e)) so that those can be exposed as well.

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.
